src/shapes/Path.spec.ts

```diff
@@ -0,0 +1,659 @@
+      'function',
+    );
+
+    const path = await Path.fromObject(REFERENCE_PATH_OBJECT);
+
+    expect(path, 'should be instance of Path').toBeInstanceOf(Path);
+    expect(path.toObject(), 'path object should match reference').toEqual(
+      REFERENCE_PATH_OBJECT,
+    );
+  });
+
+  it('fromObject with sourcePath', async () => {
+    expect(Path.fromObject, 'fromObject should be a function').toBeTypeOf(
+      'function',
+    );
+
+    const path = await Path.fromObject(REFERENCE_PATH_OBJECT);
+
+    expect(path, 'should be instance of Path').toBeInstanceOf(Path);
+    expect(path.toObject(), 'path object should match reference').toEqual(
+      REFERENCE_PATH_OBJECT,
+    );
+  });
+
+  it('fromElement', async () => {
+    expect(Path.fromElement, 'fromElement should be a function').toBeTypeOf(
+      'function',
+    );
+
+    const namespace = 'http://www.w3.org/2000/svg';
+    const elPath = getFabricDocument().createElementNS(namespace, 'path');
+
+    elPath.setAttributeNS(namespace, 'd', 'M 100 100 L 300 100 L 200 300 z');
+    elPath.setAttributeNS(namespace, 'fill', 'red');
+    elPath.setAttributeNS(namespace, 'opacity', '1');
+    elPath.setAttributeNS(namespace, 'stroke', 'blue');
+    elPath.setAttributeNS(namespace, 'stroke-width', '0');
+    elPath.setAttributeNS(namespace, 'stroke-dasharray', '5, 2');
+    elPath.setAttributeNS(namespace, 'stroke-linecap', 'round');
+    elPath.setAttributeNS(namespace, 'stroke-linejoin', 'bevel');
+    elPath.setAttributeNS(namespace, 'stroke-miterlimit', '5');
+    elPath.setAttributeNS(namespace, 'transform', 'scale(2)');
+
+    const path = await Path.fromElement(elPath);
+
+    expect(path, 'should be instance of Path').toBeInstanceOf(Path);
+    expect(path.toObject(), 'path object should match reference').toEqual({
+      ...REFERENCE_PATH_OBJECT,
+      strokeDashArray: [5, 2],
+      strokeLineCap: 'round',
+      strokeLineJoin: 'bevel',
+      strokeMiterLimit: 5,
+    });
+
+    const ANGLE_DEG = 90;
+
+    elPath.setAttributeNS(namespace, 'transform', 'rotate(' + ANGLE_DEG + ')');
+
+    const rotatedPath = await Path.fromElement(elPath);
+
+    expect(
+      rotatedPath.get('transformMatrix'),
+      'transform matrix should match expected',
+    ).toEqual([0, 1, -1, 0, 0, 0]);
+  });
+
+  it('numbers with leading decimal point', async () => {
+    expect(Path.fromElement, 'fromElement should be a function').toBeTypeOf(
+      'function',
+    );
+
+    const namespace = 'http://www.w3.org/2000/svg';
+    const elPath = getFabricDocument().createElementNS(namespace, 'path');
+
+    elPath.setAttributeNS(namespace, 'd', 'M 100 100 L 300 100 L 200 300 z');
+    elPath.setAttributeNS(namespace, 'transform', 'scale(.2)');
+
+    const path = await Path.fromElement(elPath);
+
+    expect(path, 'should be instance of Path').toBeInstanceOf(Path);
+
+    // @ts-expect-error -- TODO: check why transformMatrix is not set on the Path as potential type?
```

Comment:
@asturur should `transformMatrix` be set as nullable property of the path?

Reply:
Yeah the type should be fixed.
fromElement returns the FabricObject with attached that property that then we remove one line of code after.
Each 'FromElement' should call 'removeTransformMatrix' but i think to avoid duplicated calls it was done in a single place in the parsing logic.